x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -508,6 +509,21 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
                   },
                 });
               }
+
+              const disabledAction = rule.actions.find(
+                (action) => !actions.isActionTypeEnabled(action.actionTypeId)
+              );
+              if (disabledAction) {
+                await ruleExecutionLogger.logStatusChange({
+                  newStatus: RuleExecutionStatusEnum['partial failure'],
+                  message: `Alerts created but the registered action ${disabledAction.actionTypeId} is disabled. Please check your license / tier and ensure the connector is enabled for your given license / tier`,
```

Comment:
cc @nastasha-solomon for copy ✏️

Reply:
Hello! @dhurley14 if possible, it'd be helpful for me to see this message (and all of the other copy that you're adding) in the UI so I have a better understanding of when/where/how the copy shows up.

Reply:
@nastasha-solomon it's pretty much only showing up here in the rule details page (and also written to the rule execution log)

<img width="1199" alt="Screen Shot 2024-05-06 at 2 14 49 PM" src="https://github.com/elastic/kibana/assets/915763/15135a2c-1346-4f04-afb8-52d3a07f6f73">

Reply:
Hey, @dhurley14. I could use more information on the following:
- Is this warning only appearing in Serverless? If it is, we should only use "tier" right? Also, instead of telling users to check their level of access, can we point them towards a specific page in the app, or an external page that details the different tier options for Serverless? 
- How specific does this message need to be? Would something more general like the following work? 
   `This rule generated alerts, but did not send notifications.`
- What happens when users click Ask Assistant? Does the AI Assistant give them the same information? Or are they provided with new information?
- Does this warning persist on alerts that were generated when the connector was disabled? This might not affect copy, but it could be useful for troubleshooting docs.

Reply:
I believe that these PLI checks are specific to Serverless, is that right @dhurley14 ?

Reply:
I thought this thread would un-resolve itself if new comments were added but I guess not.. Sorry about that.

> Does this warning only appear in Serverless?

Does the alerting framework hide action connectors based on the available license? I seem to recall they do but I can't find docs on whether that's the case. If they do, then yes this can also happen with non-serverless deployments. Basically anytime an action connector is deemed "disabled" by the actions framework.

> can we point them to a specific page in the app that shows them the tier option they've been given or have?

I don't believe a page like this exists within the app. However, it is available in the project management page on cloud:
<img width="1449" alt="manage_project" src="https://github.com/elastic/kibana/assets/915763/09b2c22e-38fe-4217-bdf9-67e4f094faef">



> How specific does this message need to be? Would something more general like the following work? `This rule generated alerts, but did not send notifications.`

I think it's important to note that the notification was likely not sent because of the tier downgrade (which I believe is the only way a customer can disable a connector). I feel if the message does not include something about the license / tier then it may not be obvious to the customer why the actions aren't firing.

> What happens when users click Ask Assistant? Does the AI Assistant give them the same information? Or are they provided with new information?

Not sure how that ended up in there I think my original screenshot contains a bug (that has since been fixed in `main`) where the ai assistant button showed up when it's not available for `essentials` tier. Here is a current screenshot:
<img width="1194" alt="no_ai_assistant" src="https://github.com/elastic/kibana/assets/915763/5f389a8f-fd26-47a8-9ba1-b381910890f6">

> Does this warning persist on alerts that were generated when the connector was disabled? This might not affect copy, but it could be useful for troubleshooting docs.

I don't think we write rule execution logs into alerts generated during that time.

Thanks for the review @nastasha-solomon !

Reply:
Hey - apologies for the delay, @dhurley14. Is it possible to create separate warnings for ESS and Serverless? If we want to make clear suggestions for resolving this warning, we should only use langage that's relevant to the platform that the user is on. In other words, in the warning message, we shouldn't be telling users to check their license subscription for ESS **_or_** the Security Analytics/SIEM tier that they purchased for Serverless to confirm whether they have access to rule actions. 

If you can separate the warnings, here are some alternatives to consider:

- **ESS**: This rule generated alerts, but did not send external notifications because rule action connectors aren't enabled. To send notifications, you need a higher license.
- **Serverless**: This rule generated alerts, but did not send external notifications because rule action connectors aren't enabled. To send notifications, you need a higher Security Analytics tier.
(I'm not sure how accurate the wording for the Security tiers is, so feel free to workshop it if it's incorrect.)